src/OSPSuite.Infrastructure.Import/Core/DataFormat/AbstractColumnsDataFormat.cs

```diff
@@ -122,7 +124,7 @@ protected virtual void ExtractNonQualifiedHeadings(List<string> keys, List<strin
                );
             if (headerKey == null) continue;
             keys.Remove(headerKey);
-            var units = ExtractUnits(headerKey, data, keys, ref rank);
+            var units = ExtractUnits(headerKey, data, keys, columnInfos.First(ci => ci.DisplayName == header).SupportedDimensions.ToList(),  ref rank);
```

Comment:
I don't like this ToList here. Make sure ExtractUnits takes a IReadOnly instead of a List and you are good to go no?

Reply:
done